src/Symfony/Component/RateLimiter/RateLimiterFactory.php

```diff
@@ -68,19 +68,21 @@ public function create(?string $key = null): LimiterInterface
     protected static function configureOptions(OptionsResolver $options): void
     {
         $intervalNormalizer = static function (Options $options, string $interval): \DateInterval {
+            // Create DateTimeImmutable from unix timesatmp, so the default timezone is ignored and we don't need to
+            // deal with quirks happening when modifying dates using a timezone with DST.
+            $now = \DateTimeImmutable::createFromFormat('U', time());
+
             try {
-                // Create DateTimeImmutable from unix timesatmp, so the default timezone is ignored and we don't need to
-                // deal with quirks happening when modifying dates using a timezone with DST.
-                $now = \DateTimeImmutable::createFromFormat('U', time());
-
-                return $now->diff($now->modify('+'.$interval));
-            } catch (\Exception $e) {
-                if (!preg_match('/Failed to parse time string \(\+([^)]+)\)/', $e->getMessage(), $m)) {
-                    throw $e;
-                }
+                $nowPlusInterval = @$now->modify('+' . $interval);
 
-                throw new \LogicException(sprintf('Cannot parse interval "%s", please use a valid unit as described on https://www.php.net/datetime.formats.relative.', $m[1]));
+                if (!$nowPlusInterval) {
```

Comment:
`modify()` returns `false` on failure, thus we must not pass it to `diff()` before checking the type